bugbug/pulearning.py

```diff
@@ -0,0 +1,141 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# Module for Positive Unlabeled Learning
+import numpy as np
+from sklearn.ensemble import bagging
+from sklearn.utils import indices_to_mask
+
+
+def pu_parallel_build_estimators(
```

Comment:
Did you write this code, or did you get from somewhere? If the second, we should put a reference to the origin (and check its license to see if we can use it).

Reply:
To be exact https://github.com/roywright/pu_learning/blob/a394b2691818813441eb65270870291fbab194f9/baggingPU.py#L75

Reply:
Inspired from [https://github.com/roywright/pu_learning/blob/master/baggingPU.py](url)

Reply:
OK, it's BSD. You'll need to copy the notice from the top of that file to here.